cpp/src/arrow/array/builder_dict.cc

```diff
@@ -349,6 +355,9 @@ template class DictionaryBuilder<Date64Type>;
 template class DictionaryBuilder<Time32Type>;
 template class DictionaryBuilder<Time64Type>;
 template class DictionaryBuilder<TimestampType>;
+template class DictionaryBuilder<DurationIntervalType>;
+template class DictionaryBuilder<DayTimeIntervalType>;
+template class DictionaryBuilder<MonthIntervalType>;
```

Comment:
Just for your information.

These changes will be conflicted with #4255.
If #4255 is merged, we will be able to just remove these changes.

Reply:
Thanks, they were removed.